src/components/TabSelector/TabSelector.js

```diff
@@ -62,32 +62,46 @@ const getOpacity = (position, routesLength, tabIndex, active) => {
 
         return position.interpolate({
             inputRange,
-            outputRange: _.map(inputRange, (i) => (i === tabIndex ? activeValue : inactiveValue)),
+            outputRange: _.map(inputRange, (i) => (affectedTabs.includes(tabIndex) && i === tabIndex ? activeValue : inactiveValue)),
         });
     }
     return activeValue;
 };
 
-const getBackgroundColor = (position, routesLength, tabIndex) => {
+const getBackgroundColor = (position, routesLength, tabIndex, affectedTabs) => {
     if (routesLength > 1) {
         const inputRange = Array.from({length: routesLength}, (v, i) => i);
 
         return position.interpolate({
             inputRange,
-            outputRange: _.map(inputRange, (i) => (i === tabIndex ? themeColors.border : themeColors.appBG)),
+            outputRange: _.map(inputRange, (i) => (affectedTabs.includes(tabIndex) && i === tabIndex ? themeColors.border : themeColors.appBG)),
         });
     }
     return themeColors.border;
 };
 
 function TabSelector({state, navigation, onTabPress, position}) {
     const {translate} = useLocalize();
+    const [, reRender] = useReducer((x) => !x, false);
```

Comment:
This looks kinda hack-ish. Wouldn't it be better to just keep `affectedAnimatedTabs` in state like below?

```
const defaultAffectedAnimatedTabs = Array.from({length: state.routes.length}, (v, i) => i);
const [affectedAnimatedTabs, setAffectedAnimatedTabs] = React.useState(defaultAffectedAnimatedTabs);

React.useEffect(() => {
    setTimeout(() => {
        setAffectedAnimatedTabs(defaultAffectedAnimatedTabs);
    }, CONST.ANIMATED_TRANSITION);
    // eslint-disable-next-line react-hooks/exhaustive-deps
}, [state.index]);
```

Reply:
@burczu 
Yes, I think this is better.
Updated!